paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/client/impl/ClientImpl.java

```diff
@@ -104,7 +110,16 @@ private ClientImpl(final ClientImpl.Builder builder) {
     @Override
     public Single<String> getEvent(final EventTypeCursor cursor) {
         final Observable<HttpResponseChunk> request = getContent0(cursor, 1);
-        return request.map(chunk -> getEvent0(chunk.getContent())).firstOrDefault(null).toSingle();
+        try {
+            return request.map(chunk -> getEvent0(chunk.getContent(), cursor.getEventType())).firstOrDefault(null)
+                          .toSingle();
+        } catch (InvalidEventTypeException e) {
+            LOGGER.error("Invalid event type error for offset [{}] on partition [{}]: {}", cursor.getOffset(),
```

Comment:
The idea here is to have the offset and partition in the log without the need to pass it down to the EventUtils, but still have the exception thrown.